plugins/network/smtp_hello_

```diff
@@ -20,37 +17,227 @@
+    SMTP_COMMAND    - SMTP command to use to test the connection. Default is "HELO localhost". For some mail servers it
+                      may be required to change this to e.g. "QUIT" or "HELP"
+    TIMEOUT_SECONDS - Timeout for the SMTP connection test. default is 120 (seconds)
+
+=back
+
+Example:
+
+=over 2
+
+    [smtp_hello_*]
+    env.SMTP_PORTS 25 465 587
+    env.SMTP_COMMAND HELO localhost
+    env.TIMEOUT_SECONDS 60
+
+=back
+
+Default warnings and criticals are as following:
+Emit a warning when we take more than half the configured timeout time (60s). Set to critical when we get more than 99%
+ of the timeout time (118s). Those are calculated as an integer, for timeouts with lower numbers also configure individual
+ warnings and criticals. The plugin also emits a warning when curl is not able to communicate properly with the SMTP
+ server.
+
+Different warnings and critical can be specified as following:
+
+=over 2
+
+    [smtp_hello_*]
+    # warn when any port doesn't respond within 20 seconds (note make sure to overwrite this for status)
+    env.warning :10
+    # warn when port 25 doesn't respond within 20 seconds
+    env.host_warning :20
+    # critical when port 25 doesn't respond within 60 seconds
+    env.host_critical :60
+    # any status code that is not zero should be a warning
+    env.status_warning 0:0
+    # warn when port 25 doesn't respond within 40 seconds
+    env.host_465_warning :40
+    # critical when port 465 doesn't respond within 60 seconds
+    env.host_465_critical :60
+    # any status code that is not zero should be a warning
+    env.status_465_warning 0:0
+
+=back
+
+
+ [postfix_mailqueue]
+    env.spooldir /var/spool/postfix
+
+=head1 AUTHOR
+
+Marek Mahut
+
+Copyright (C) 2008 Marek Mahut <mmahut@fedoraproject.org>
+Copyright (C) 2022 Andreas Perhab, WT-IO-IT GmbH <a.perhab@wtioit.at>
+
+=head1 LICENSE
+
+Gnu GPLv2
+
+SPDX-License-Identifier: GPL-2.0-or-later
 
-        response=`echo HELO localhost | /usr/bin/time -f "%e" nc -C -w 120 $host 25 2>&1 | tail -n 1`
-        echo "host.value $response"
 
+
+=head1 MAGIC MARKERS
+
+=begin comment
+
+These magic markers are used by munin-node-configure when installing
+munin-node.
+
+=end comment
+
+ #%# family=auto
+ #%# capabilities=autoconf
+
+=cut
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+if [ "${MUNIN_DEBUG:-0}" == "1" ]; then
+  set -x
+fi
+
+SMTP_COMMAND=${SMTP_COMMAND:-HELO localhost}
+SMTP_PORTS=${SMTP_PORTS:-25}
+TIMEOUT_SECONDS=${TIMEOUT_SECONDS:-120}
+
+host=$(basename "$0" | sed 's/^smtp_hello_//g')
+
+time_executable=""
+if which time >/dev/null; then
+  # use time executable found in PATH
+  time_executable=$(which time)
+elif [ -x /usr/bin/time ]; then
+  # use time executable found in /usr/bin
+  time_executable="/usr/bin/time"
+fi
+if [ "$time_executable" != "" ]; then
+  function take_the_time() {
+    "$time_executable" -f "%e" "$@" 2>&1 | tail -n 1
+    return "${PIPESTATUS[0]}"
+  }
+elif (time echo) >/dev/null 2>&1; then
```

Comment:
I think, you can omit the brackets here. At least I do not see a reason for spawning a sub-shell.